homeassistant/components/leviosa_shades/__init__.py

```diff
@@ -0,0 +1,40 @@
+"""The Leviosa shades Zone integration."""
+import logging
+
+from leviosapy import LeviosaZoneHub as tZoneHub
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [Platform.COVER]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Leviosa shades Zone from a config entry."""
+    hub_name = entry.title
+    hub_ip = entry.data[CONF_HOST]
+    try:
+        hub = tZoneHub(
+            hub_ip=hub_ip, hub_name=hub_name, websession=async_get_clientsession(hass)
+        )
+        await hub.getHubInfo()  # Check all is good
+    except Exception as err:
+        raise ConfigEntryError("get hub info failed") from err
```

Comment:
I know that `catch Exception` is a very ugly way of writing.
 In fact, the initial version does not need to try to catch anything, because the constructor of `LeviosaZoneHub` and the method `getHubInfo` should handle all exceptions themselves.
My understanding is that `async_setup_entry` in `__init__.py` has already obtained the IP of the hub. If communication with this IP fails thereafter, then we have no way to determine whether the IP of the hub has changed or the hub is not powered on. Therefore, there is no way to distinguish between `ConfigEntryNotReady` and `ConfigEntryAuthFailed`.

Reply:
Suggestion accepted, thanks.